src/hevm/hevm-cli/hevm-cli.hs

```diff
@@ -593,24 +596,38 @@ dappCoverage opts _ solcFile =
         let
           dapp = dappInfo "." contractMap sourceCache
           f (k, vs) = do
-            putStr ("\x1b[0m" ++ "————— hevm coverage for ") -- Prefixed with color reset
-            putStrLn (unpack k ++ " —————")
-            putStrLn ""
-            forM_ vs $ \(n, bs) -> do
-              case ByteString.find (\x -> x /= 0x9 && x /= 0x20 && x /= 0x7d) bs of
-                Nothing -> putStr "\x1b[38;5;240m" -- Gray (Coverage status isn't relevant)
-                Just _ ->
-                  case n of
-                    -1 -> putStr "\x1b[38;5;240m" -- Gray (Coverage status isn't relevant)
-                    0  -> putStr "\x1b[31m" -- Red (Uncovered)
-                    _  -> putStr "\x1b[32m" -- Green (Covered)
-              Char8.putStrLn bs
-            putStrLn ""
-
+            if shouldPrintCoverage (EVM.UnitTest.covMatch opts) k then do
+              putStr ("\x1b[0m" ++ "————— hevm coverage for ") -- Prefixed with color reset
+              putStrLn (unpack k ++ " —————")
+              putStrLn ""
+              forM_ vs $ \(n, bs) -> do
+                case ByteString.find (\x -> x /= 0x9 && x /= 0x20 && x /= 0x7d) bs of
+                  Nothing -> putStr "\x1b[38;5;240m" -- Gray (Coverage status isn't relevant)
+                  Just _ ->
+                    case n of
+                      -1 -> putStr "\x1b[38;5;240m" -- Gray (Coverage status isn't relevant)
+                      0  -> putStr "\x1b[31m" -- Red (Uncovered)
+                      _  -> putStr "\x1b[32m" -- Green (Covered)
+                Char8.putStrLn bs
+              putStrLn ""
+            else pure ()
         mapM_ f (Map.toList (coverageReport dapp covs))
       Nothing ->
         error ("Failed to read Solidity JSON for `" ++ solcFile ++ "'")
 
+shouldPrintCoverage :: Maybe Text -> Text -> Bool
+shouldPrintCoverage (Just covMatch) file = regexMatches covMatch file
+shouldPrintCoverage Nothing file = not (isCoverageIrrelevant file)
+
+isCoverageIrrelevant :: Text -> Bool
+isCoverageIrrelevant file = isAnyPrefixOf ["src/test/", "src/tests/", "lib/"] file || Text.isSuffixOf ".t.sol" file
+
+isAnyPrefixOf :: [Text] -> Text -> Bool
+isAnyPrefixOf [] _ = False
+isAnyPrefixOf (x:xs) t
+  | Text.isPrefixOf x t = True
+  | otherwise = isAnyPrefixOf xs t
```

Comment:
better way than guards to do this? 

or could this whole function be refactored into some sort of fold or map operation?

Reply:
Yeah, you could do:
```suggestion
isAnyPrefixOf prefixes t = any (flip Text.isPrefixOf t) prefixes
```

Reply:
Or smth like this

Reply:
oo dope